### ISSUE TITLE
Not All Expected Interceptor Cookies are Being Sent

### ISSUE THREAD

**Issue by hereiznoy** [a85/POSTMan-Chrome-Extension#9541]
**Describe the Issue**

Cookies being captured by Interceptor at the `domain.com` level are not being sent by Postman in requests to `sub.domain.com`. (Any cookies set at the `sub.domain.com` level are sent as expected.) Cookies at both levels are present when inspected via the *Manage Cookies* dialogue. But when inspecting the sent request (I used Fiddler), only the cookies for `sub.domain.com` are sent. This started happening after upgrading from v7.x to v8.x.

**Steps to Reproduce**

1. Set up Interceptor to grab cookies for `domain.com`
2. Send a request to `sub.domain.com`
3. Inspect request, see that cookies for `sub.domain.com` are sent but not the cookies for `domain.com`

**Environment Information**

 - Platform Type: Native Desktop App
 - Postman Version: 8.0.4
 - Operating System: Windows 10
 - Browser (Interceptor): Chromium Edge Beta 88.0.705.56

**Comment by kotot** [comment]
postman 8.0.3 8.0.4 , have same issue,  just send cookie of sub.domain.com, v7.36.1 is correct
Operating System: mac os 11.2
Postman Version: 8.0.4

**Comment by codenirvana** [comment]
@hereiznoy @kotot Can you share steps to reproduce this issue?

Here's what I tried (without Interceptor):
* Request `domain.com`, responds with `Set-Cookie: foo=bar; Path=/; Domain=domain.com`
* Request `domain.com` again, cookie `foo` is sent
* Request `sub.domain.com`, cookie `foo` is sent

Is this happening only with Interceptor?

**Comment by hereiznoy** [comment]
@codenirvana Yes, only happening with Interceptor. As in, the browser has the cookie(s) I need sent with the request I am sending from Postman. So expected behavior:

1. Acquire cookies in the browser (where Interceptor is installed and connected) for `domain.com` and `sub.domain.com`
2. Send request to `sub.domain.com` from Postman
3. See cookies from both `domain.com` and `sub.domain.com` sent with request

But I'm only seeing the `sub.domain.com` cookies sent in the request. These aren't cookies that can be set as part of an initial request from Postman, due to some nasty JS redirects standing in the way, so they either have to come from Interceptor, or I have to add them manually in Postman.

**Comment by codenirvana** [comment]
@hereiznoy Thanks for the information. Let's scope down this problem.

Can you share the exact cookie string (from Cookies manager) of one of the `domain.com` cookie which was supposed to be sent with `sub.domain.com` request?

e.g, `foo=bar; Path=/; Domain=domain.com; Secure; Expires=Sat, 06 Mar 2021 15:55:41 GMT;`

_Remove all sensitivity information from the cookie string._

**Comment by hereiznoy** [comment]
@codenirvana So in the cookie manager, the cookie string value is: `foo=bar; path=/;`.  The `Domain` key-value pair is not specified, as I would expect it to be (the cookie itself is under the **domain.com** grouping in the cookie manager). If I edit that cookie in the cookie manager to include `Domain=.domain.com`, the cookie gets sent with the request to `sub.domain.com` as expected. 

If I send a request from the browser and inspect via Fiddler, I see the _Set-Cookie_ value includes `Domain=.domain.com` in the cookie string, so prior to being picked up by Interceptor, the string explicitly specified Domain.

**Comment by codenirvana** [comment]
@hereiznoy You're right, cookies imported via Interceptor are missing the `Domain` attribute making them HostOnly.

Thanks for reporting this, will let you know once this issue is fixed.

**Comment by Rydgel** [comment]
I confirm having this issue since upgrading from 7.x to 8.x.
I usually use a collection pre-script like that to set the value of a CSRF cookie to a variable.

```js
pm.sendRequest({
    url: "http://api.mydomain.local/api/csrf-cookie",
    method: "GET"
}, function (error, response, { cookies }) {
    if (!error) {
        pm.environment.set('xsrf-cookie', cookies.get('XSRF-TOKEN'));
    }
})
```

This doesn't work anymore, even with the cookie domain being `.mydomain.local`.

EDIT: to be more precise `cookies` is empty even tho it's returned in the response header.

**Comment by vinitshahdeo** [comment]
Hello @hereiznoy, @Rydgel & @kotot,

The latest [Postman Interceptor](https://chrome.google.com/webstore/detail/postman-interceptor/aicmkgpgakddgnaphhhpliifpcfhicfo?hl=en) `v1.1.2` contains fix for this issue.  Can you please update your extension and confirm?

**Comment by hereiznoy** [comment]
@vinitshahdeo I can confirm that updating to `v1.1.2` of Interceptor resolves the issue for me. Thanks!

**Comment by levinsamuel** [comment]
@vinitshahdeo It does not seem to be working for me. My interceptor version is `1.1.2`, updated March 1. Postman `v8.0.6`.

The way it works at my company is we get a session cookie for `env.company.com`. And I'm sending a request to `myapp.env.company.com`. I can see the `env.company.com` cookie in my cookie list in Postman. However, I still am unauthorized when I send the request to myapp. I can tell the cookie is not being sent by looking at the "Code" of the request. It looks like this:

```
curl --location --request GET 'https://myapp.env.company.com/... \
--data-raw ''
```
And I'm still unauthorized. When I add the cookie to my specific domain in Postman, I see the cookie in the request, and the request works.

```
curl --location --request GET 'https://myapp.env.company.com/... \
--header 'Cookie: cookie-key=<cookie-value>' \
--data-raw ''
```

**Comment by vinitshahdeo** [comment]
@levinsamuel  Assuming that the expected session cookie isn't hostOnly cookie, can you please share the exact cookie string(after removing sensitive info) of `env.company.com` which was supposed to be sent with `myapp.env.company.com` request?

e.g, `foo=bar; Path=/; Domain=env.company.com; Secure; Expires=Sat, 27 Mar 2021 15:55:41 GMT;`

You may try out the following troubleshooting steps:

1. Turn `Capture cookies` off.
2. Delete all your cookies for `env.company.com` domain  from `Manage Cookies` modal. (Refer [this](https://learning.postman.com/docs/sending-requests/cookies/) link)
3. Turn `Capture cookies` on.

**Comment by levinsamuel** [comment]
So, the first thing is I know this worked for me before Postman 8. 

This is what shows up in Postman when I open that cookie:
```
cookie-name=<cookie-value>; Path=/; Secure;
```
This is what I see in the response headers when I get that cookie back from the browser, when I log in:
```
set-cookie: cookie-name=<cookie-value>; Domain=.env.company.com; expires=Sat, 06 Mar 2021 20:04:25 GMT; HttpOnly; Max-Age=14400; Path=/; Secure
```
Meaning the cookie is not host-only https://stackoverflow.com/questions/18492576/share-cookie-between-subdomain-and-domain. This works as expected in the browser, where I use that cookie to access my app on the subdomain.

No, doing those steps doesn't work. My cookies seem to be picked up correctly by the interceptor, I can always see the `env.company.com` cookie in the list. The problem is Postman doesn't seem to be attaching it when I send a request to the sub-domain, even though the cookie is not supposed to be host-only.

EDIT: adding the `Domain` key in postman makes it work again, so I manually edit that cookie in postman to be:
```
cookie-name=<cookie-value>; Path=/; Secure; Domain=env.company.com;
```
And that also makes it work. So it seems as if the "Domain" property isn't getting saved in Postman even if it comes with the cookie. Could that have something to do with the leading `.` we use?

**Comment by lanzelotik** [comment]
I have this issue in Postman Version 8.6.2. For both cases (with inspector and with manually added cookies)

**Steps to Reproduce**

1. Add new GET request http://my.domain.org/testurl
2. Open Manage Cookies and add cookie for domain domain.org
3. Request was sent without cookie
4. Change request url to http://domain.org/testurl
5. Cookie was sent as expected

**Comment by lanzelotik** [comment]
UPD. 
By default cookie creates without domain attribute, so after I change value to
`Cookie_5=value; Path=/; Domain=domain.org; Expires=Fri, 17 Jun 2022 13:23:52 GMT;` 
It works correctly

But.
For domain name `domain.local` I still have the issue. So the additional issue is with .local top domain.

**Comment by LoLFactor** [comment]
Can confirm issue on current version (`8.6.2`), and it seems @lanzelotik nailed it. I was also using a `.local` domain for developing an application locally.

**Setup:**
 - Server setting cookies running on `<level2>.<domain>.local`
 - Cookie being set with domain attribute `.<domain>.local`

**Behaviour:**
 - Postman receives cookie with proper domain attribute (`.<domain>.local`) and saves it, i.e. it shows up in "Manage cookies" (side-note: the leading dot always gets truncated, despite the `Set-Cookie` header having it)
 - Postman does not send cookie header on subsequent request to `<level2>.<domain>.local`

**Expected behaviour:**
 - Cookie should be sent, especially if it includes a leading dot
 
 Upon reading the previous comment, I changed the setup to replace the `.local` part with `.dev`. Problem goes away. Postman behaves as expected.
